birdclef/utils.py

```diff
@@ -56,17 +56,47 @@ def load_audio(input_path: Path, offset: float, duration: int = 7, sr: int = 320
     return np.resize(np.moveaxis(y_trunc, -1, 0), length)
 
 
-def slice_seconds(data, sample_rate, seconds=5, pad_seconds=0):
+def slice_seconds(
+    data, sample_rate, seconds=5, pad_seconds=0, remainder_padding_type="center"
+):
+    valid_padding_types = ["center", "right"]
```

Comment:
One last implementation please. Do you think you could add in an implementation for "right-aligned" (I can't think a good phrasing) where the last element of the array is just `data[-step:-1]`  i.e. the slice of length step that's right aligned to the end of the audio?

Reply:
Does this just mean you want 3 options total?
- Right-pad (just add zeros to the end of the audio until it divides evenly into the step size)
- Center-pad (add an even number of zeros to both the start and end of the audio so that the audio itself is centered and divides evenly)
- Left-pad (this is the new one -- just add zeros to the beginning of the audio until it divides evenly into the step size)

Reply:
No, left padding would make sense here logically following the existing options, but the "right-aligned" padding option starts from the end and moves left. If the last slice is index n, then it will likely overlap with index n-1.

Reply:
```
input: [1, 2, 3, 4, 5]
# seconds=2, right pad
[[1, 2], [3, 4], [5, 0]]
# seconds=2, right-aligned
[1, 2], [3, 4], [4, 5]

# and to clear up my intention with center
input: [1, 2, 3, 4, 5, 6, 7]
# seconds=3, center
[0, 1, 2], [3, 4, 5], [6, 7, 0]
```

Reply:
Hmm, still trying to make sure I understand it.  Does the "right-aligned" method create duplication of some data elements between the last slice `n` and the slice before it `n-1`?

For example, if the input array is `[1, 2, 3, 4, 5, 6]` with a step size of 4, does this mean you expect the result to be `[ [1, 2, 3, 4], [3, 4, 5, 6] ]`?  So the last slice (which previously would have been truncated) is instead a full slice without padding, but it may re-sample some overlapping elements in order to fill an entire slice-window?

Reply:
It would indeed cause duplication between windows. However, the triplet creation should be relatively robust wrt to duplication like this (since we're only looking for neighbors and not an exact position) and my gut feeling is that it's better than throwing the partial data point. The competition dataset only has audio that exactly fits 1 minute, so we want the extracted slices to have full clips of audio.